src/modules/string_utils.rs

```diff
@@ -162,6 +162,84 @@ impl StringUtils {
         })
     }
 
+    // $at (S n -- S')
+    #[cmd(name = "$at", stack)]
+    fn interpret_str_at(stack: &mut Stack) -> Result<()> {
+        let index = stack.pop_usize()?;
+        let string = stack.pop_string()?;
+
+        match string.chars().nth(index) {
+            Some(s) => stack.push(s.to_string()),
+            None => anyhow::bail!("index must be >= 0 and <= {}", usize::MAX),
```

Comment:
`index` range is guaranteed to be `0..usize::MAX`, it is better to change error to `Index out of string bounds`

Reply:
FIXED